WORKSPACE

```diff
@@ -61,30 +69,85 @@ rules_proto_toolchains()
 # section
 http_archive(
     name = "com_google_api_codegen",
-    strip_prefix = "gapic-generator-2.7.0",
-    urls = ["https://github.com/googleapis/gapic-generator/archive/v2.7.0.zip"],
+    strip_prefix = "gapic-generator-2.9.0",
+    urls = ["https://github.com/googleapis/gapic-generator/archive/v2.9.0.zip"],
 )
 
+# rules_go (support Golang under bazel)
+# This is not in the Go section because we override the same, older dependency brought in by gRPC.
+# TODO(ndietz): move this back to the Go section if gRPC is updated per https://github.com/grpc/grpc/issues/22172
 http_archive(
-    name = "com_google_disco_to_proto3_converter",
-    urls = ["https://github.com/googleapis/disco-to-proto3-converter/archive/b2fffa8a43bbe84eeb65541f3fde30c859f00f0e.zip"],
-    strip_prefix = "disco-to-proto3-converter-b2fffa8a43bbe84eeb65541f3fde30c859f00f0e"
+    name = "io_bazel_rules_go",
+    sha256 = "d1ffd055969c8f8d431e2d439813e42326961d0942bdf734d2c95dc30c369566",
+    urls = [
+        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.24.5/rules_go-v0.24.5.tar.gz",
+        "https://github.com/bazelbuild/rules_go/releases/download/v0.24.5/rules_go-v0.24.5.tar.gz",
+    ],
 )
 
-load("@com_google_disco_to_proto3_converter//:repository_rules.bzl", "com_google_disco_to_proto3_converter_properties")
+load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
 
-com_google_disco_to_proto3_converter_properties(
-    name = "com_google_disco_to_proto3_converter_properties",
-    file = "@com_google_disco_to_proto3_converter//:pom.xml",
+go_rules_dependencies()
+
+go_register_toolchains()
+
+##############################################################################
+# C++
+##############################################################################
+# C++ must go before everything else, since the key dependencies (protobuf and gRPC)
```

Comment:
minor: Well, not "everything else" since you have stuff that must go above. Maybe say "Aside from the special considerations above, the rules for C++ must go before the main rules for the other languages...."